### PR TITLE
Feature/ij 75 journal entries view

### PR DESCRIPTION
**Pull request by IeuanSkinner**
Team members can now view a paginated index of all the journal entries that have been created **that they have an associated permissions record for.** Once a team member then views the journal entry a view log is created and this is observable for them so they can keep track of what they have/haven't read.

The full index of these view logs can be read by admins by looking at the team members member view as is the case for the WBA Self View Logs records.

Team member dashboard includes 'Journal Entries' button that links to paginated index.
Screenshot 2021-03-17 at 17 32 08

Index shows a team member journal entries they are permitted to see through the journal entry permissions record. Viewed status is also shown.
Screenshot 2021-03-17 at 17 32 14

Journal entry member record
Screenshot 2021-03-17 at 17 32 20

### CODE REVIEW

**Comment by phil-6** [comment]
@AlexW555  and @bmhtech07  please can you check out the branch and explore the UI to see what you think.

File: app/views/team_members/team_members/show.html.erb [hunk 1]
```diff
@@ -48,6 +48,22 @@
             </div>
           <% end %>
         </div>
+        <div class="row">
+          <a class="col col d-flex align-items-center" style="background-color: transparent; border-color: transparent"></a>
+          <a class="col col d-flex align-items-center" style="background-color: transparent; border-color: transparent"></a>
```
Comment (phil-6) [inline app/views/team_members/team_members/show.html.erb:53]: Please can we extract the inline styles?

File: app/views/users/journal_entries/index.html.erb [hunk 2]
```diff
@@ -1,4 +1,4 @@
-<main class="container user journal-entry index">
+<main class="container user journal_entry index">
```
Comment (phil-6) [inline app/views/users/journal_entries/index.html.erb:1]: I've been down quite a rabbit hole on this. I think CSS classes should be hyphen-delimited not underscore_delimited or camelCase. 

https://cssguidelin.es/#hyphen-delimited

File: app/views/team_members/journal_entry_view_logs/show.html.erb [hunk 3]
```diff
@@ -15,10 +15,10 @@
         <th scope="col">Created At</th>
```
Comment (phil-6) [inline app/views/team_members/journal_entry_view_logs/show.html.erb:15]: h3 title needs updating

File: app/views/team_members/journal_entries/show.html.erb [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+      <h3 class="mb-3">Journal Entry</h3>
+      <h5 class="mb-3"><%= link_to "#{@journal_entry.user.first_name} #{@journal_entry.user.last_name}", user_path(@journal_entry.user) %>
+        created this entry <%= time_ago_in_words(@journal_entry.created_at) %> ago</h5>
+      <div class="card mb-3">
+        <div class="card-header">
+          <span><%= @journal_entry.feeling %></span>
+          <span><strong>Created At:</strong> <%= @journal_entry.created %></span>
+        </div>
+        <div class="form-floating">
+          <textarea class="form-control" id="journal_entry" disabled><%= @journal_entry.entry %></textarea>
```
Comment (phil-6) [inline app/views/team_members/journal_entries/show.html.erb:20]: This text area has a resize tag but doesn't seem to let you resize it.

Peek 2021-03-18 08-24 no resize

**Review by phil-6 — COMMENTED** [review]
Overall this looks good. As per the other PR the back button seems to break occasionally, but I do _really_ like the back buttons. They're a great addition

I'm a little bit concerned about the way that pagination works at the moment, only in relation to building search functionality in the future. Fine for now but probably something to just do some background thinking about. 

In the view logs pages, do you think it is worth including some identifier to say what specific item they've viewed. 
Maybe change the current "created at" column to "viewed at" and add a column for "journal created at" or "assessment created at" which links to that specific assessment or view log.

File: app/controllers/team_members/journal_entries_controller.rb [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    before_action :count, :last_page, :journal_entry_view_logs, only: :page_index, if: -> { @journal_entries.present? }
+    before_action :redirect, only: :page_index, unless: -> { @journal_entries.present? }
+
+    before_action :journal_entry, :add_view_log, only: :show
+
+    LIMIT = 5
+
+    # GET /journal_entries
+    def index
+      redirect_to pages_journal_entries_path(1)
```
[the remark this answers is not in the document]
Comment (IeuanSkinner) [reply app/controllers/team_members/journal_entries_controller.rb:14]: First page of paginated index = 1

**Comment by IeuanSkinner** [comment]
I'm closing this for now so I can do more work on it.

**Pull request closed without merging**